### PR TITLE
[Enhancement] Replace react-data-grid with react-virtualized

### PR DESCRIPTION
**Pull request by macrigiuseppe**
I previously tried to replace react-data-grid using react-window but unfortunately it didn't work out 
the way I thought (#562).

Features:
- Replaces react-data-grid with more performant react-virtualized MultiGrid
- DataGrid (Data display table) is now a factory and DataTableModalFactory dependency to allow developers to customize look & feel data table modal.
- DataGridFactory has now two dependencies:
 - FieldHeaderFactory: renders DataGrid Header
 - CellFactory: renders Cell

data-table

Signed-off-by: Giuseppe Macri <gmacri@uber.com>

### CODE REVIEW

File: src/components/modals/data-table-modal.js [hunk 1]
```diff
@@ -20,74 +20,114 @@
...
-  const {navigator} = window;
-  // Detect browsers
-  // http://stackoverflow.com/questions/5899783/detect-safari-using-jquery
-  const isMac = navigator.userAgent.match(/Macintosh/);
-  const is_chrome = navigator.userAgent.indexOf('Chrome') > -1;
-  const is_safari = navigator.userAgent.indexOf('Safari') > -1;
-  const is_firefox = navigator.userAgent.indexOf('Firefox') > -1;
-
-  // prevent chrome scroll back
-  shouldPreventScrollBack = isMac && (is_chrome || is_safari || is_firefox);
```
Comment (macrigiuseppe) [inline src/components/modals/data-table-modal.js]: no longer necessary, the table component takes care of handling mouse wheel

File: src/components/modals/data-table-modal.js [hunk 2]
```diff
@@ -20,74 +20,114 @@
...
-}
+import AutoSizer from 'react-virtualized/dist/commonjs/AutoSizer'
```
Comment (macrigiuseppe) [inline src/components/modals/data-table-modal.js]: The missing piece in this PR is the dynamic size of the table, which autosizer may provide but i am trying to see if there are other options

File: src/components/modals/data-table-modal.js [hunk 3]
```diff
@@ -158,21 +189,18 @@ export class DataTableModal extends Component {
           datasets={datasets}
           showDatasetTable={showDatasetTable}
         />
-        <DataGridWrapper
-          onWheel={shouldPreventScrollBack ? this._onMouseWheel : null}
```
Comment (macrigiuseppe) [inline src/components/modals/data-table-modal.js]: no longer necessary

File: src/components/modals/data-table-modal.js (hunk 2 above)
Comment (macrigiuseppe) [reply src/components/modals/data-table-modal.js]: Removed Autosizer completely, disregard this one;

**Comment by macrigiuseppe** [comment]
@heshan0131 ready to review

File: src/components/common/datagrid.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+// THE SOFTWARE.
+
+import React, {PureComponent} from 'react';
+import {MultiGrid} from 'react-virtualized';
+import styled from 'styled-components';
+import FieldToken from 'components/common/field-token';
+import {Clock} from 'components/common/icons/index';
+import {ALL_FIELD_TYPES} from 'constants/default-settings';
+
+const COLUMN_WIDTH = 200;
```
Comment (heshan0131) [inline src/components/common/datagrid.js]: Can this be put into theme?

**Comment by heshan0131** [comment]
One small things:
Can you add `outline: 0` to css `:focus, :active` selector so that this blue outline doesnt show when click in the table body?

Screen Shot 2019-07-28 at 10 16 36 PM

File: src/components/common/datagrid.js (hunk 4 above)
Comment (macrigiuseppe) [reply src/components/common/datagrid.js]: done

**Comment by macrigiuseppe** [comment]
> One small things:
> Can you add `outline: 0` to css `:focus, :active` selector so that this blue outline doesnt show when click in the table body?
> 
> Screen Shot 2019-07-28 at 10 16 36 PM

@heshan0131 remove outline

**Pull request merged**